src/topology/vector_bundle/basic.lean

```diff
@@ -45,312 +45,230 @@ Vector bundle
 noncomputable theory
 
 open bundle set
-open_locale classical
+open_locale classical bundle
 
 variables (R 𝕜 : Type*) {B : Type*} (F : Type*) (E : B → Type*)
 
 section topological_vector_space
-variables [semiring R] [∀ x, add_comm_monoid (E x)] [∀ x, module R (E x)]
-  [topological_space F] [add_comm_monoid F] [module R F] [topological_space B]
+variables {B F E} [semiring R]
+  [topological_space F]  [topological_space B]
 
-/-- A pretrivialization for a (yet to be defined) topological vector bundle `total_space E` is a
-local equiv between sets of the form `proj ⁻¹' base_set` and `base_set × F` which respects the
-first coordinate, and is linear in each fiber. -/
-@[ext, nolint has_nonempty_instance]
-structure topological_vector_bundle.pretrivialization extends to_fiber_bundle_pretrivialization :
-  topological_fiber_bundle.pretrivialization F (@total_space.proj B E) :=
-(linear' : ∀ x ∈ base_set, is_linear_map R (λ y : E x, (to_fun (total_space_mk x y)).2))
+/-- A mixin class for `pretrivialization`, stating that a pretrivialization is fibrewise linear with
+respect to given module structures on its fibres and the model fibre. -/
+protected class pretrivialization.is_linear [add_comm_monoid F] [module R F]
+  [∀ x, add_comm_monoid (E x)] [∀ x, module R (E x)] (e : pretrivialization F (π E)) :
+  Prop :=
+(linear : ∀ b ∈ e.base_set, is_linear_map R (λ x : E b, (e (total_space_mk b x)).2))
 
-instance : has_coe_to_fun (topological_vector_bundle.pretrivialization R F E) _ := ⟨λ e, e.to_fun⟩
+namespace pretrivialization
 
-instance : has_coe (topological_vector_bundle.pretrivialization R F E)
-  (topological_fiber_bundle.pretrivialization F (@total_space.proj B E)) :=
-⟨topological_vector_bundle.pretrivialization.to_fiber_bundle_pretrivialization⟩
+variables {F E} (e : pretrivialization F (π E)) {x : total_space E} {b : B} {y : E b}
 
-namespace topological_vector_bundle.pretrivialization
+lemma linear [add_comm_monoid F] [module R F] [∀ x, add_comm_monoid (E x)] [∀ x, module R (E x)]
+  [e.is_linear R] {b : B} (hb : b ∈ e.base_set) :
+  is_linear_map R (λ x : E b, (e (total_space_mk b x)).2) :=
+pretrivialization.is_linear.linear b hb
 
-open topological_vector_bundle
-
-variables {R F E} (e : pretrivialization R F E) {x : total_space E} {b : B} {y : E b}
-
-protected lemma linear (hb : b ∈ e.base_set) :
-  is_linear_map R (λ y : E b, (e (total_space_mk b y)).2) :=
-e.linear' b hb
-
-@[simp, mfld_simps] lemma coe_coe : ⇑e.to_local_equiv = e := rfl
-@[simp, mfld_simps] lemma coe_fst (ex : x ∈ e.source) : (e x).1 = x.proj := e.proj_to_fun x ex
-lemma mem_source : x ∈ e.source ↔ x.proj ∈ e.base_set := by rw [e.source_eq, mem_preimage]
 lemma coe_mem_source : ↑y ∈ e.source ↔ b ∈ e.base_set := e.mem_source
```

Comment:
I certainly intend to in a follow-up PR, but I had left it as-is here so that the diff was easier to review.  Would you indeed prefer to have it all done in the same PR?

Reply:
I'm perfectly fine with the move being done in another PR, so let's keep it like that for now.